### DISCUSSION TITLE
Can't install the control package

### DISCUSSION THREAD

**Discussion by NoraYV** [python-control/python-control#706] [Q&A] [answered]
Hey
I work with Anaconda and PyCharme. I tried to install the control package but get an error _RemoveError: 'setuptools' is a dependency of conda and cannot be removed from conda's operating environment._ My whole code from the anaconda powershell prompt is liested below. How do I fix this error and be able to use the control package?

`(base) PS C:\Users\Yazdandoost> conda install -c conda-forge control 
Collecting package metadata (current_repodata.json): done
Solving environment: \
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - defaults/noarch::alabaster==0.7.12=pyhd3eb1b0_0
  - defaults/win-64::anaconda==2021.11=py39_0
  - defaults/win-64::anaconda-client==1.9.0=py39haa95532_0
  - defaults/win-64::anaconda-navigator==2.1.1=py39_0
  - defaults/noarch::anaconda-project==0.10.1=pyhd3eb1b0_0
  - defaults/win-64::anyio==2.2.0=py39haa95532_2
  - defaults/noarch::appdirs==1.4.4=pyhd3eb1b0_0
  - defaults/win-64::argh==0.26.2=py39haa95532_0
  - defaults/win-64::argon2-cffi==20.1.0=py39h2bbff1b_1
  - defaults/win-64::arrow==0.13.1=py39haa95532_0
  - defaults/noarch::asn1crypto==1.4.0=py_0
  - defaults/win-64::astropy==4.3.1=py39hc7d831d_0
  - defaults/noarch::async_generator==1.10=pyhd3eb1b0_0
  - defaults/noarch::atomicwrites==1.4.0=py_0
  - defaults/noarch::attrs==21.2.0=pyhd3eb1b0_0
  - defaults/noarch::autopep8==1.5.7=pyhd3eb1b0_0
  - defaults/noarch::babel==2.9.1=pyhd3eb1b0_0
  - defaults/noarch::backcall==0.2.0=pyhd3eb1b0_0
  - defaults/noarch::backports==1.0=pyhd3eb1b0_2
  - defaults/noarch::backports.functools_lru_cache==1.6.4=pyhd3eb1b0_0
  - defaults/noarch::backports.shutil_get_terminal_size==1.0.0=pyhd3eb1b0_3
  - defaults/noarch::backports.tempfile==1.0=pyhd3eb1b0_1
  - defaults/noarch::backports.weakref==1.0.post1=py_1
  - defaults/win-64::bcrypt==3.2.0=py39h196d8e1_0
  - defaults/noarch::beautifulsoup4==4.10.0=pyh06a4308_0
  - defaults/noarch::binaryornot==0.4.4=pyhd3eb1b0_1
  - defaults/win-64::bitarray==2.3.0=py39h2bbff1b_1
  - defaults/win-64::bkcharts==0.2=py39haa95532_0
  - defaults/noarch::black==19.10b0=py_0
  - defaults/noarch::bleach==4.0.0=pyhd3eb1b0_0
  - defaults/win-64::bokeh==2.4.1=py39haa95532_0
  - defaults/win-64::boto==2.49.0=py39haa95532_0
  - defaults/win-64::bottleneck==1.3.2=py39h7cc1a96_1
  - defaults/win-64::brotlipy==0.7.0=py39h2bbff1b_1003
  - defaults/noarch::cached-property==1.5.2=py_0
  - defaults/win-64::certifi==2021.10.8=py39haa95532_0
  - defaults/win-64::cffi==1.14.6=py39h2bbff1b_0
  - defaults/win-64::chardet==4.0.0=py39haa95532_1003
  - defaults/noarch::charset-normalizer==2.0.4=pyhd3eb1b0_0
  - defaults/noarch::click==8.0.3=pyhd3eb1b0_0
  - defaults/noarch::cloudpickle==2.0.0=pyhd3eb1b0_0
  - defaults/win-64::clyent==1.2.2=py39haa95532_1
  - defaults/noarch::colorama==0.4.4=pyhd3eb1b0_0
  - defaults/win-64::comtypes==1.1.10=py39haa95532_1002
  - defaults/win-64::conda==4.11.0=py39haa95532_0
  - defaults/win-64::conda-build==3.21.6=py39haa95532_0
  - defaults/noarch::conda-content-trust==0.1.1=pyhd3eb1b0_0
  - defaults/noarch::conda-pack==0.6.0=pyhd3eb1b0_0
  - defaults/win-64::conda-package-handling==1.7.3=py39h8cc25b3_1
  - defaults/noarch::conda-repo-cli==1.0.4=pyhd3eb1b0_0
  - defaults/noarch::conda-token==0.3.0=pyhd3eb1b0_0
  - defaults/noarch::conda-verify==3.4.2=py_1
  - defaults/win-64::console_shortcut==0.1.1=4
  - defaults/noarch::contextlib2==0.6.0.post1=pyhd3eb1b0_0
  - conda-forge/noarch::control==0.9.1=pyhd8ed1ab_2
  - defaults/noarch::cookiecutter==1.7.2=pyhd3eb1b0_0
  - defaults/win-64::cryptography==3.4.8=py39h71e12ea_0
  - defaults/win-64::cycler==0.10.0=py39haa95532_0
  - defaults/win-64::cython==0.29.24=py39h604cdb4_0
  - defaults/win-64::cytoolz==0.11.0=py39h2bbff1b_0
  - defaults/win-64::daal4py==2021.3.0=py39h757b272_0
  - defaults/noarch::dask==2021.10.0=pyhd3eb1b0_0
  - defaults/noarch::dask-core==2021.10.0=pyhd3eb1b0_0
  - defaults/noarch::dataclasses==0.8=pyh6d0b6a4_7
  - defaults/win-64::debugpy==1.4.1=py39hd77b12b_0
  - defaults/noarch::decorator==5.1.0=pyhd3eb1b0_0
  - defaults/noarch::defusedxml==0.7.1=pyhd3eb1b0_0
  - defaults/noarch::diff-match-patch==20200713=pyhd3eb1b0_0
  - defaults/win-64::distributed==2021.10.0=py39haa95532_0
  - defaults/win-64::docutils==0.17.1=py39haa95532_1
  - defaults/win-64::entrypoints==0.3=py39haa95532_0
  - defaults/win-64::et_xmlfile==1.1.0=py39haa95532_0
  - defaults/win-64::fastcache==1.1.0=py39h196d8e1_0
  - defaults/noarch::filelock==3.3.1=pyhd3eb1b0_1
  - defaults/noarch::flake8==3.9.2=pyhd3eb1b0_0
  - defaults/noarch::flask==1.1.2=pyhd3eb1b0_0
  - defaults/noarch::fonttools==4.25.0=pyhd3eb1b0_0
  - defaults/noarch::fsspec==2021.10.1=pyhd3eb1b0_0
  - defaults/win-64::future==0.18.2=py39haa95532_1
  - defaults/win-64::get_terminal_size==1.0.0=h38e98db_0
  - defaults/win-64::gevent==21.8.0=py39h2bbff1b_1
  - defaults/noarch::glob2==0.7=pyhd3eb1b0_0
  - defaults/win-64::greenlet==1.1.1=py39hd77b12b_0
  - defaults/win-64::h5py==3.2.1=py39h3de5c98_0
  - defaults/noarch::heapdict==1.0.1=pyhd3eb1b0_0
  - defaults/noarch::html5lib==1.1=pyhd3eb1b0_0
  - defaults/noarch::idna==3.2=pyhd3eb1b0_0
  - defaults/win-64::imagecodecs==2021.8.26=py39ha1f97ea_0
  - defaults/noarch::imageio==2.9.0=pyhd3eb1b0_0
  - defaults/noarch::imagesize==1.2.0=pyhd3eb1b0_0
  - defaults/win-64::importlib-metadata==4.8.1=py39haa95532_0
  - defaults/noarch::importlib_metadata==4.8.1=hd3eb1b0_0
  - defaults/win-64::inflection==0.5.1=py39haa95532_0
  - defaults/noarch::iniconfig==1.1.1=pyhd3eb1b0_0
  - defaults/noarch::intervaltree==3.1.0=pyhd3eb1b0_0
  - defaults/win-64::ipykernel==6.4.1=py39haa95532_1
  - defaults/win-64::ipython==7.29.0=py39hd4e2768_0
  - defaults/noarch::ipython_genutils==0.2.0=pyhd3eb1b0_1
  - defaults/noarch::ipywidgets==7.6.5=pyhd3eb1b0_1
  - defaults/noarch::isort==5.9.3=pyhd3eb1b0_0
  - defaults/noarch::itsdangerous==2.0.1=pyhd3eb1b0_0
  - defaults/noarch::jdcal==1.4.1=pyhd3eb1b0_0
  - defaults/win-64::jedi==0.18.0=py39haa95532_1
  - defaults/noarch::jinja2==2.11.3=pyhd3eb1b0_0
  - defaults/noarch::jinja2-time==0.2.0=pyhd3eb1b0_2
  - defaults/noarch::joblib==1.1.0=pyhd3eb1b0_0
  - defaults/noarch::json5==0.9.6=pyhd3eb1b0_0
  - defaults/noarch::jsonschema==3.2.0=pyhd3eb1b0_2
  - defaults/win-64::jupyter==1.0.0=py39haa95532_7
  - defaults/noarch::jupyterlab==3.2.1=pyhd3eb1b0_1
  - defaults/noarch::jupyterlab_pygments==0.1.2=py_0
  - defaults/noarch::jupyterlab_server==2.8.2=pyhd3eb1b0_0
  - defaults/noarch::jupyterlab_widgets==1.0.0=pyhd3eb1b0_1
  - defaults/noarch::jupyter_client==6.1.12=pyhd3eb1b0_0
  - defaults/noarch::jupyter_console==6.4.0=pyhd3eb1b0_0
  - defaults/win-64::jupyter_core==4.8.1=py39haa95532_0
  - defaults/win-64::jupyter_server==1.4.1=py39haa95532_0
  - defaults/win-64::keyring==23.1.0=py39haa95532_0
  - defaults/win-64::kiwisolver==1.3.1=py39hd77b12b_0
  - defaults/win-64::lazy-object-proxy==1.6.0=py39h2bbff1b_0
  - defaults/win-64::llvmlite==0.37.0=py39h23ce68f_1
  - defaults/win-64::locket==0.2.1=py39haa95532_1
  - defaults/win-64::lxml==4.6.3=py39h9b66d53_0
  - defaults/win-64::markupsafe==1.1.1=py39h2bbff1b_0
  - defaults/win-64::matplotlib==3.4.3=py39haa95532_0
  - defaults/win-64::matplotlib-base==3.4.3=py39h49ac443_0
  - defaults/noarch::matplotlib-inline==0.1.2=pyhd3eb1b0_2
  - defaults/win-64::mccabe==0.6.1=py39haa95532_1
  - defaults/win-64::menuinst==1.4.18=py39h59b6b97_0
  - defaults/win-64::mistune==0.8.4=py39h2bbff1b_1000
  - defaults/win-64::mkl-service==2.4.0=py39h2bbff1b_0
  - defaults/win-64::mkl_fft==1.3.1=py39h277e83a_0
  - defaults/win-64::mkl_random==1.2.2=py39hf11a4ad_0
  - defaults/noarch::mock==4.0.3=pyhd3eb1b0_0
  - defaults/noarch::more-itertools==8.10.0=pyhd3eb1b0_0
  - defaults/win-64::mpmath==1.2.1=py39haa95532_0
  - defaults/win-64::msgpack-python==1.0.2=py39h59b6b97_1
  - defaults/win-64::multipledispatch==0.6.0=py39haa95532_0
  - defaults/noarch::munkres==1.1.4=py_0
  - defaults/win-64::mypy_extensions==0.4.3=py39haa95532_0
  - defaults/win-64::navigator-updater==0.2.1=py39haa95532_0
  - defaults/noarch::nbclassic==0.2.6=pyhd3eb1b0_0
  - defaults/noarch::nbclient==0.5.3=pyhd3eb1b0_0
  - defaults/win-64::nbconvert==6.1.0=py39haa95532_0
  - defaults/noarch::nbformat==5.1.3=pyhd3eb1b0_0
  - defaults/noarch::nest-asyncio==1.5.1=pyhd3eb1b0_0
  - defaults/noarch::networkx==2.6.3=pyhd3eb1b0_0
  - defaults/noarch::nltk==3.6.5=pyhd3eb1b0_0
  - defaults/noarch::nose==1.3.7=pyhd3eb1b0_1006
  - defaults/win-64::notebook==6.4.5=py39haa95532_0
  - defaults/win-64::numba==0.54.1=py39hf11a4ad_0
  - defaults/win-64::numexpr==2.7.3=py39hb80d3ca_1
  - defaults/win-64::numpy==1.20.3=py39ha4e8547_0
  - defaults/win-64::numpy-base==1.20.3=py39hc2deb75_0
  - defaults/noarch::numpydoc==1.1.0=pyhd3eb1b0_1
  - defaults/noarch::olefile==0.46=pyhd3eb1b0_0
  - defaults/noarch::openpyxl==3.0.9=pyhd3eb1b0_0
  - defaults/noarch::packaging==21.0=pyhd3eb1b0_0
  - defaults/win-64::pandas==1.3.4=py39h6214cd6_0
  - defaults/win-64::pandocfilters==1.4.3=py39haa95532_1
  - defaults/noarch::paramiko==2.7.2=py_0
  - defaults/noarch::parso==0.8.2=pyhd3eb1b0_0
  - defaults/noarch::partd==1.2.0=pyhd3eb1b0_0
  - defaults/win-64::path==16.0.0=py39haa95532_0
  - defaults/noarch::path.py==12.5.0=hd3eb1b0_0
  - defaults/win-64::pathlib2==2.3.6=py39haa95532_2
  - defaults/noarch::pathspec==0.7.0=py_0
  - defaults/win-64::patsy==0.5.2=py39haa95532_0
  - defaults/win-64::pep8==1.7.1=py39haa95532_0
  - defaults/noarch::pexpect==4.8.0=pyhd3eb1b0_3
  - defaults/noarch::pickleshare==0.7.5=pyhd3eb1b0_1003
  - defaults/win-64::pillow==8.4.0=py39hd45dc43_0
  - defaults/win-64::pip==21.2.4=py39haa95532_0
  - defaults/win-64::pkginfo==1.7.1=py39haa95532_0
  - defaults/win-64::pluggy==0.13.1=py39haa95532_0
  - defaults/win-64::ply==3.11=py39haa95532_0
  - defaults/win-64::powershell_shortcut==0.0.1=3
  - defaults/noarch::poyo==0.5.0=pyhd3eb1b0_0
  - defaults/noarch::prometheus_client==0.11.0=pyhd3eb1b0_0
  - defaults/noarch::prompt-toolkit==3.0.20=pyhd3eb1b0_0
  - defaults/noarch::prompt_toolkit==3.0.20=hd3eb1b0_0
  - defaults/win-64::psutil==5.8.0=py39h2bbff1b_1
  - defaults/noarch::ptyprocess==0.7.0=pyhd3eb1b0_2
  - defaults/noarch::py==1.10.0=pyhd3eb1b0_0
  - defaults/win-64::py-lief==0.10.1=py39hd77b12b_1
  - defaults/noarch::pycodestyle==2.7.0=pyhd3eb1b0_0
  - defaults/win-64::pycosat==0.6.3=py39h2bbff1b_0
  - defaults/noarch::pycparser==2.20=py_2
  - defaults/win-64::pycurl==7.44.1=py39hcd4344a_1
  - defaults/noarch::pydocstyle==6.1.1=pyhd3eb1b0_0
  - defaults/win-64::pyerfa==2.0.0=py39h2bbff1b_0
  - defaults/noarch::pyflakes==2.3.1=pyhd3eb1b0_0
  - defaults/noarch::pygments==2.10.0=pyhd3eb1b0_0
  - defaults/win-64::pyjwt==2.1.0=py39haa95532_0
  - defaults/win-64::pylint==2.9.6=py39haa95532_1
  - defaults/noarch::pyls-spyder==0.4.0=pyhd3eb1b0_0
  - defaults/win-64::pynacl==1.4.0=py39hbd8134f_1
  - defaults/win-64::pyodbc==4.0.31=py39hd77b12b_0
  - defaults/noarch::pyopenssl==21.0.0=pyhd3eb1b0_1
  - defaults/noarch::pyparsing==3.0.4=pyhd3eb1b0_0
  - defaults/win-64::pyqt==5.9.2=py39hd77b12b_6
  - defaults/win-64::pyreadline==2.1=py39haa95532_1
  - defaults/win-64::pyrsistent==0.18.0=py39h196d8e1_0
  - defaults/win-64::pysocks==1.7.1=py39haa95532_0
  - defaults/win-64::pytables==3.6.1=py39h56d22b6_1
  - defaults/win-64::pytest==6.2.4=py39haa95532_2
  - defaults/win-64::python==3.9.7=h6244533_1
  - defaults/noarch::python-dateutil==2.8.2=pyhd3eb1b0_0
  - defaults/noarch::python-libarchive-c==2.9=pyhd3eb1b0_1
  - defaults/noarch::python-lsp-black==1.0.0=pyhd3eb1b0_0
  - defaults/noarch::python-lsp-jsonrpc==1.0.0=pyhd3eb1b0_0
  - defaults/noarch::python-lsp-server==1.2.4=pyhd3eb1b0_0
  - defaults/noarch::python-slugify==5.0.2=pyhd3eb1b0_0
  - conda-forge/win-64::python_abi==3.9=2_cp39
  - defaults/noarch::pytz==2021.3=pyhd3eb1b0_0
  - defaults/win-64::pywavelets==1.1.1=py39h080aedc_4
  - defaults/win-64::pywin32==228=py39he774522_0
  - defaults/win-64::pywin32-ctypes==0.2.0=py39haa95532_1000
  - defaults/win-64::pywinpty==0.5.7=py39haa95532_0
  - defaults/win-64::pyyaml==6.0=py39h2bbff1b_1
  - defaults/win-64::pyzmq==22.2.1=py39hd77b12b_1
  - defaults/noarch::qdarkstyle==3.0.2=pyhd3eb1b0_0
  - defaults/noarch::qstylizer==0.1.10=pyhd3eb1b0_0
  - defaults/noarch::qtawesome==1.0.2=pyhd3eb1b0_0
  - defaults/noarch::qtconsole==5.1.1=pyhd3eb1b0_0
  - defaults/noarch::qtpy==1.10.0=pyhd3eb1b0_0
  - defaults/win-64::regex==2021.8.3=py39h2bbff1b_0
  - defaults/noarch::requests==2.26.0=pyhd3eb1b0_0
  - defaults/noarch::rope==0.19.0=pyhd3eb1b0_0
  - defaults/win-64::rtree==0.9.7=py39h2eaa2aa_1
  - defaults/win-64::ruamel_yaml==0.15.100=py39h2bbff1b_0
  - defaults/win-64::scikit-image==0.18.3=py39hf11a4ad_0
  - defaults/win-64::scikit-learn==0.24.2=py39hf11a4ad_1
  - defaults/win-64::scikit-learn-intelex==2021.3.0=py39haa95532_0
  - defaults/win-64::scipy==1.7.1=py39hbe87c03_2
  - defaults/noarch::seaborn==0.11.2=pyhd3eb1b0_0
  - defaults/noarch::send2trash==1.8.0=pyhd3eb1b0_1
  - defaults/win-64::simplegeneric==0.8.1=py39haa95532_2
  - defaults/noarch::singledispatch==3.7.0=pyhd3eb1b0_1001
  - defaults/win-64::sip==4.19.13=py39hd77b12b_0
  - defaults/noarch::six==1.16.0=pyhd3eb1b0_0
  - conda-forge/win-64::slycot==0.4.0.0=py39h6635163_8
  - defaults/win-64::sniffio==1.2.0=py39haa95532_1
  - defaults/noarch::snowballstemmer==2.1.0=pyhd3eb1b0_0
  - defaults/noarch::sortedcollections==2.1.0=pyhd3eb1b0_0
  - defaults/noarch::sortedcontainers==2.4.0=pyhd3eb1b0_0
  - defaults/noarch::soupsieve==2.2.1=pyhd3eb1b0_0
  - defaults/noarch::sphinx==4.2.0=pyhd3eb1b0_1
  - defaults/win-64::sphinxcontrib==1.0=py39haa95532_1
  - defaults/noarch::sphinxcontrib-applehelp==1.0.2=pyhd3eb1b0_0
  - defaults/noarch::sphinxcontrib-devhelp==1.0.2=pyhd3eb1b0_0
  - defaults/noarch::sphinxcontrib-htmlhelp==2.0.0=pyhd3eb1b0_0
  - defaults/noarch::sphinxcontrib-jsmath==1.0.1=pyhd3eb1b0_0
  - defaults/noarch::sphinxcontrib-qthelp==1.0.3=pyhd3eb1b0_0
  - defaults/noarch::sphinxcontrib-serializinghtml==1.1.5=pyhd3eb1b0_0
  - defaults/noarch::sphinxcontrib-websupport==1.2.4=py_0
  - defaults/win-64::spyder==5.1.5=py39haa95532_1
  - defaults/win-64::spyder-kernels==2.1.3=py39haa95532_0
  - defaults/win-64::sqlalchemy==1.4.22=py39h2bbff1b_0
  - defaults/win-64::statsmodels==0.12.2=py39h2bbff1b_0
  - defaults/win-64::sympy==1.9=py39haa95532_0
  - defaults/win-64::tbb4py==2021.4.0=py39h59b6b97_0
  - defaults/noarch::tblib==1.7.0=pyhd3eb1b0_0
  - defaults/win-64::terminado==0.9.4=py39haa95532_0
  - defaults/noarch::testpath==0.5.0=pyhd3eb1b0_0
  - defaults/noarch::text-unidecode==1.3=pyhd3eb1b0_0
  - defaults/noarch::textdistance==4.2.1=pyhd3eb1b0_0
  - defaults/noarch::threadpoolctl==2.2.0=pyh0d69192_0
  - defaults/noarch::three-merge==0.1.1=pyhd3eb1b0_0
  - defaults/noarch::tifffile==2021.7.2=pyhd3eb1b0_2
  - defaults/noarch::tinycss==0.4=pyhd3eb1b0_1002
  - defaults/noarch::toml==0.10.2=pyhd3eb1b0_0
  - defaults/noarch::toolz==0.11.1=pyhd3eb1b0_0
  - defaults/win-64::tornado==6.1=py39h2bbff1b_0
  - defaults/noarch::tqdm==4.62.3=pyhd3eb1b0_1
  - defaults/noarch::traitlets==5.1.0=pyhd3eb1b0_0
  - defaults/win-64::typed-ast==1.4.3=py39h2bbff1b_1
  - defaults/noarch::typing_extensions==3.10.0.2=pyh06a4308_0
  - defaults/win-64::ujson==4.0.2=py39hd77b12b_0
  - defaults/win-64::unicodecsv==0.14.1=py39haa95532_0
  - defaults/noarch::unidecode==1.2.0=pyhd3eb1b0_0
  - defaults/noarch::urllib3==1.26.7=pyhd3eb1b0_0
  - defaults/win-64::watchdog==2.1.3=py39haa95532_0
  - defaults/noarch::wcwidth==0.2.5=pyhd3eb1b0_0
  - defaults/win-64::webencodings==0.5.1=py39haa95532_1
  - defaults/noarch::werkzeug==2.0.2=pyhd3eb1b0_0
  - defaults/noarch::wheel==0.37.0=pyhd3eb1b0_1
  - defaults/noarch::whichcraft==0.6.1=pyhd3eb1b0_0
  - defaults/win-64::widgetsnbextension==3.5.1=py39haa95532_0
  - defaults/win-64::wincertstore==0.2=py39haa95532_2
  - defaults/win-64::win_inet_pton==1.1.0=py39haa95532_0
  - defaults/win-64::win_unicode_console==0.5=py39haa95532_0
  - defaults/win-64::wrapt==1.12.1=py39h196d8e1_1
  - defaults/noarch::xlrd==2.0.1=pyhd3eb1b0_0
  - defaults/noarch::xlsxwriter==3.0.1=pyhd3eb1b0_0
  - defaults/win-64::xlwings==0.24.9=py39haa95532_0
  - defaults/win-64::xlwt==1.3.0=py39haa95532_0
  - defaults/noarch::xmltodict==0.12.0=pyhd3eb1b0_0
  - defaults/noarch::yapf==0.31.0=pyhd3eb1b0_0
  - defaults/noarch::zict==2.0.0=pyhd3eb1b0_0
  - defaults/noarch::zipp==3.6.0=pyhd3eb1b0_0
  - defaults/win-64::zope==1.0=py39haa95532_1
  - defaults/win-64::zope.event==4.5.0=py39haa95532_0
  - defaults/win-64::zope.interface==5.4.0=py39h2bbff1b_0
  - defaults/win-64::_ipyw_jlab_nb_ext_conf==0.1.0=py39haa95532_0
done

## Package Plan ##

  environment location: C:\Users\Yazdandoost\Anaconda3

  added / updated specs:
    - control

The following NEW packages will be INSTALLED:

  _anaconda_depends  pkgs/main/win-64::_anaconda_depends-2021.11-py39_0
  astroid            pkgs/main/win-64::astroid-2.6.6-py39haa95532_0
  setuptools         conda-forge/win-64::setuptools-60.9.3-py39hcbf5309_0

The following packages will be SUPERSEDED by a higher-priority channel:

  ca-certificates    pkgs/main::ca-certificates-2021.10.26~ --> conda-forge::ca-certificates-2021.10.8-h5b45459_0
  openssl              pkgs/main::openssl-1.1.1l-h2bbff1b_0 --> conda-forge::openssl-1.1.1l-h8ffe710_0

The following packages will be DOWNGRADED:

  anaconda                                   2021.11-py39_0 --> custom-py39_1

Proceed ([y]/n)? y

**Preparing transaction: done
Verifying transaction: failed

RemoveError: 'setuptools' is a dependency of conda and cannot be removed from
conda's operating environment.**
`

**Comment by murrayrm** [comment, marked as answer]
I've noticed that I sometimes get conflicts when I try to install packages from conda-forge on top of packages that were installed from the default conda channel.  My fix has been to install everything from conda-forge (so use `-c conda-forge` for all packages when you set up the environment.

As a quick check, you could set up a separate conda environment and see if you have problems.  Something like
```
conda create -n python3.9-control python=3.9
conda activate python3.9-control
conda install -c conda-forge control
```
(I'm not sure how to do that in PyCharme, but presumably fairly straightforward.)

**Reply by NoraYV** [reply]
Thank you! After creating a new environment, install first the control packages and afterwards the other packages it works without any conflicts!
